### PR TITLE
lds: pause SRDS when LDS is updated

### PR DESCRIPTION
**Pull request by tony612**
Commit Message: lds: pause SRDS when LDS is updated
Additional Description:

Now RDS is paused when LDS is updated, so also pause SRDS in this PR. Though I didn't hit the related problem, I guess we also should do this?

btw, I didn't see an existing test for pausing RDS in LDS, so didn't update the test. If one exists somewhere, please point it out and I can update it for SRDS. Or I can try to add a new one if needed.

Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: source/extensions/listener_managers/listener_manager/lds_api.cc [hunk 1]
```diff
@@ -46,6 +47,7 @@ void LdsApiImpl::onConfigUpdate(const std::vector<Config::DecodedResourceRef>& a
   if (cm_.adsMux()) {
     const std::vector<std::string> paused_xds_types{
         Config::getTypeUrl<envoy::config::route::v3::RouteConfiguration>(),
+        Config::getTypeUrl<envoy::config::route::v3::ScopedRouteConfiguration>(),
```
[the remark this answers is not in the document]
Comment (tony612) [reply source/extensions/listener_managers/listener_manager/lds_api.cc:50]: I guess so. AFAIK, `maybe_resume_rds_sds = cm_.adsMux()->pause(paused_xds_types)` will pause the XDS. And when `maybe_resume_rds_sds` is dropped after `onConfigUpdate` is finished, the XDS will be resumed.

[the remark this answers is not in the document]
Comment (adisuissa) [reply source/extensions/listener_managers/listener_manager/lds_api.cc:50]: Yep, the ScopedResume object ensures that the SRDS requests will not be sent until it is destroyed (end of the method in this case).

**Review by adisuissa — COMMENTED** [review]
The change LGTM, Thanks!
Please add a test. See test added in #21600 as an inspiration.

**Comment by adisuissa** [comment]
Waiting on a test.
/wait

File: test/integration/ads_integration_test.cc [hunk 2]
```diff
@@ -2367,4 +2367,98 @@ TEST_P(XdsTpAdsIntegrationTest, EdsAlternatingLedsUsage) {
...
+  test_server_->waitForCounterEq("listener_manager.listener_added", 2);
+
+  // Expect a single request for SRDS resources.
+  EXPECT_TRUE(compareDiscoveryRequest(srds_type_url, "", {}, {}, {}));
+
+  EXPECT_TRUE(compareDiscoveryRequest(lds_type_url, "1", {}, {}, {}));
+
+  std::vector<envoy::config::route::v3::ScopedRouteConfiguration> scoped_route_configs;
+  for (int i = 0; i < 2; ++i) {
+    envoy::config::route::v3::ScopedRouteConfiguration scoped_route;
```
[the remark this answers is not in the document]
Comment (tony612) [reply test/integration/ads_integration_test.cc:2442]: Ah, missed this somehow. Added now and pushed with other fixes.

**Review by zuercher — APPROVED** [review]

**Comment by tony612** [comment]
Merged main to resolve the changelog conflict.

**Review by zuercher — APPROVED** [review]

**Pull request merged**